src/Sql/Platform/IbmDb2/SelectDecorator.php

```diff
@@ -82,6 +104,20 @@ protected function processLimitOffset(PlatformInterface $platform, DriverInterfa
             return;
         }
 
+        if ($this->supportsLimitOffset) {
+            // Note: db2_prepare/db2_execute fails with positional parameters, for LIMIT & OFFSET
+            $limit = (int)$this->limit;
+            $offset = (int)$this->offset;
+            if ($this->limit) {
```

Comment:
`$this->limit` is not casted to `(int)` one, I think should use `$limit` instead